kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/clients/records/BaseConsumerRecord.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.clients.records;
+
+import java.util.Optional;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.header.Headers;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.kroxylicious.systemtests.utils.KafkaUtils;
+
+public class BaseConsumerRecord {
+    private static final Logger LOGGER = LoggerFactory.getLogger(BaseConsumerRecord.class);
+
+    protected String topic;
+    protected long timestamp;
+    protected String timestampType;
```

Comment:
do the tests ever care about timestamp type?   I wouldn't both deserialising fields you don't need.